py/kubeflow/testing/new_test_util.py

```diff
@@ -0,0 +1,155 @@
+"""
+This module contains misc utils used for Kubeflow testing
+infrastructure. They include creating test cases/suites,
+managing logs of test cases and initializing tests
+
+This is intended to replace test_util.py
+"""
+import argparse
+import logging
+import os
+import subprocess
+import time
+import xml.etree.ElementTree as ET
+
+import junit_xml
+
+
+class TestCase(junit_xml.TestCase):
+  """A single test case."""
+
+  def __init__(self, test_func, **kwargs):
+    self.class_name = "Kubeflow"
+    self.start_time = time.time()
+    self.test_func = test_func
+    super(TestCase, self).__init__(**kwargs)
+
+  def add_failure_info(self, message=None, output=None, failure_type=None):
+    if message:
+      if not self.failure_message:
+        self.failure_message = ''
+      self.failure_message += '\n' + message
+    if output:
+      if not self.failure_output:
+        self.failure_output = ''
+      self.failure_output += '\n' + output
+    if failure_type:
+      self.failure_type = failure_type
+
+class TestSuite(junit_xml.TestSuite):
+  """A suite of test cases."""
+
+  def __init__(self, name, artifacts_dir, logs_dir, **kwargs):
+    self.artifacts_dir = artifacts_dir
+    self.logs_dir = logs_dir
+    super(TestSuite, self).__init__(name, **kwargs)
+
+  def generate_xml(self):
+    output_file = os.path.join(self.artifacts_dir,
+                               "junit_" + self.name + ".xml")
```

Comment:
I remember we should replace "_" by "-" (or the opposite) in the name?

Reply:
I tested `_` and it seems to work.

The Gubernator example https://github.com/kubernetes/test-infra/tree/master/gubernator also uses `_`

Reply:
See #638.
https://github.com/kubeflow/kubeflow/pull/638/files#diff-2552a6a0cb5a74e3184790464405bedcR309

test-grid is the problem and we want only 1 "_" in the name.

Please fix and a comment explaining why we do this.